### ISSUE TITLE
Accessibility: Comment indicators aren't readable by assistive technologies

### ISSUE THREAD

**Issue by aparkening** [alistapart/AListApart#220]
From Andrew's accessibility audit:

In comments, the text showing the number of the comment or whether the comment is from an ALA’er is not able to be read by assistive technologies.

  Using CSS ::before and content: attr(data-count); to populate numbers and “staff” icon.  Neither is expressed in content.
![comment-indicators](https://cloud.githubusercontent.com/assets/7241111/7903703/8b6e0560-07b2-11e5-95ed-e8e392eafd91.png)

These issues represent WCAG 2.0 1.3.1 Failures

**Comment by murtaugh** [comment]
@awkawk Does that fact that this is an ordered list not mitigate that?

**Comment by awkawk** [comment]
The use of the OL mitigates the need for the "3" in the above screenshot, but not for the ALA indicator.  For that one an assistive technology user would know that it is the 6th comment, but the missing information is that the comment comes from an official ALA person.

**Comment by murtaugh** [comment]
Very good point. We've often said it might be a too-subtle indicator ever for sighted readers, so t's something we've been intending to revisit at some point.

**Comment by murtaugh** [comment]
@sarawb: I (and I think @awkawk) would like to suggest adding a text-based indication that an author has composed a particular comment. At the moment we do what is above — namely show a laurel instead of a number, and darken the comment's border.

What do you think of adding something like "Rick Pastoor (article author) 2:24 am on July 15, 2015"?

**Comment by aparkening** [comment]
@murtaugh: are you suggesting something like "Rick Pastoor (article author) 2:24 am on July 15, 2015" as the full title that everyone sees or as an alt or title attribute?

**Comment by awkawk** [comment]
I think that the indication is clear visually (although authors and ALA'ers have the same iconography) so this can be handled with aria-label.  We are already using "commenter name" with aria-label (and VoiceOver is doing something funky with that which I need to look into) so perhaps we could do some differentiation along those lines.

**Comment by murtaugh** [comment]
Some people don't think the current visual indicator is enough, so yeah I think "(article author)" or something might be helpful. If that conflicts with an aria role, we could remove the role, no?

**Comment by awkawk** [comment]
Right now the aria-labels in use in this area aren't doing what they should.  The label is not being read at all, just the link text.  If looking to augment the current information I'd do this:

<p class="comment-byline">
	<span class="commenter-name" aria-label="Article Author" id="aaa">
	   <a href="http://perpendicularangel.com"id="bbb" aria-labelledby="bbb aaa">Anne Gibson</a>
        </span>
	<time datetime="1441723131" aria-label="comment posted on" id="ccc"><a href="/comments/designing-safer-web-animation-for-motion-sensitivity#339531" aria-labelledby="ccc ddd" id="ddd">10:38 am on September 08, 2015</a></time>
</p>

**Comment by murtaugh** [comment]
But I'd also like to add something visual, so we wouldn't really need this, right?

**Comment by awkawk** [comment]
possibly for the date if we still want to augment that part

**Comment by murtaugh** [comment]
This is turning out to be trouble — I'd forgotten that the logic for the User addon makes this sort of thing stupidly tricky. I've asked Michelle to look into some PHP I can't get to work, for unknown reasons.

**Comment by murtaugh** [comment]
Michelle spotted what I was doing wrong. I've committed some updates, but they won't be available to test until dev is showing the right branch.

In the meantime, here's my proposal for a new display for an author's comments:

![screen shot 2015-09-15 at 6 26 12 pm](https://cloud.githubusercontent.com/assets/308752/9894372/da3aaac6-5bd7-11e5-953b-06604a80d957.png)

**Comment by awkawk** [comment]
Tim, I'm wondering if the "(article author)" will be part of the link with the name or with timestamp.  I think that it should be part of the name.  I wonder whether the styling should indicate that more explicitly to avoid mouse users clicking on the first part of the bunch of light grey text to get a link to the specific comment and being confused why they are at the author bio.

**Comment by murtaugh** [comment]
It's not part of either link, actually. I don't think it needs to be, and, IMO, the parenthesis set it apart enough from the date stamp that I'm not concerned about confusion.

@mikepick should take a look at it in any case.

**Comment by awkawk** [comment]
I don't think that it needs to be part of the link with the author name, so long as it isn't linked at all.  So I think that we agree there.  As far as whether the parentheses set it off enough I don't have a strong opinion.

**Comment by mikepick** [comment]
This solution seems fine to me.
